### ISSUE TITLE
Cannot click_on a link or button with aria-label in a Rails app

### ISSUE THREAD

**Issue by cbothner** [teamcapybara/capybara#2125]
Capybara won’t find and click on a button or a link with aria-label, even with `Capybara.enable_aria_label = true`

## Meta
Capybara Version:
```Using capybara 3.10.0```

Driver Information (and browser if relevant):
```driven_by :selenium, using: :chrome, screen_size: [1400, 1400]```

## Failing test script

```ruby
# frozen_string_literal: true

require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem 'capybara'
  gem 'rails', '5.2.0'
  gem 'webdrivers'
end

require 'rack/test'
require 'action_controller/railtie'

class TestApp < Rails::Application
  config.root = __dir__
  config.session_store :cookie_store, key: 'cookie_store_key'
  secrets.secret_key_base = 'secret_key_base'

  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger

  routes.draw do
    get '/' => 'test#index'
  end
end

class TestController < ActionController::Base
  include Rails.application.routes.url_helpers

  def index
    render inline: <<~HTML
      <a href="#" aria-label="Link">icon</a>
      <button aria-label="Button">icon</button>
    HTML
  end
end

Capybara.server = :webrick
Capybara.enable_aria_label = true

require 'minitest/autorun'

class BugTest < ActionDispatch::SystemTestCase
  driven_by :selenium, using: :chrome, screen_size: [1400, 1400]

  def test_click_on_link_by_aria_label
    visit '/'
    assert_selector '[aria-label="Link"]'
    click_on 'Link'
  end

  def test_click_on_button_by_aria_label
    visit '/'
    assert_selector '[aria-label="Button"]'
    click_on 'Button'
  end

  private

  def app
    Rails.application
  end
end

=begin
Error:
BugTest#test_click_on_link_by_aria_label:
Capybara::ElementNotFound: Unable to find link or button "Link"

Error:
BugTest#test_click_on_button_by_aria_label:
Capybara::ElementNotFound: Unable to find link or button "Button"
=end
```

**Comment by twalpole** [comment]
Looks like `aria-label` isn't support in the :link_or_button_selector (which is used by `click_on`) -- it will work if you use `click_link` or `click_button` respectively

**Comment by twalpole** [comment]
Hmmm -- although I have no idea why -- it's just combining the separate `link` and `button` selectors so it should be supported

**Comment by twalpole** [comment]
Ok -- I believe this is fixed in the `link_or_button_aria` branch -- waiting for tests to pass

**Comment by twalpole** [comment]
@cbothner You're welcome -- thank you for the clear issue report with code to replicate -- makes things MUCH easier 👍
